### DISCUSSION TITLE
Unhandled Runtime Error Error: PrismaClient is unable to run in this browser environment, or has been bundled for the browser (running in `unknown`). If this is unexpected, please open an issue: https://pris.ly/prisma-prisma-bug-report

### DISCUSSION THREAD

**Discussion by veyis** [prisma/prisma#24279] [Q&A]
### Bug description

Unhandled Runtime Error
Error: PrismaClient is unable to run in this browser environment, or has been bundled for the browser (running in `unknown`).
If this is unexpected, please open an issue: https://pris.ly/prisma-prisma-bug-report

### Expected behavior

    const posts = await prisma.post.findMany();

### Prisma information

```prisma
// Add your schema.prisma
```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: 
- Database: 
- Node.js version: 

### Prisma Version

```

```

**Comment by ludralph** [comment]
Hi @veyis 👋 

The error message you're seeing, `PrismaClient is unable to run in this browser environment, or has been bundled for the browser (running in unknown)`, is a common issue when trying to use Prisma in a browser environment or in a context that is not supported by Prisma.

Prisma is designed to run in a server-side environment and it does not support running directly in the browser. This is because Prisma needs to connect to your database, which should not be exposed directly to the client-side for security reasons.

If you're seeing this error in a Next.js application, it could be because you're trying to use Prisma in the client-side part of your application. Prisma should only be used in server-side parts of your application, such as API routes.

You can take a look at this closed [issue](https://github.com/prisma/prisma/issues/6219) to see how other users solved the same error.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
